xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -918,16 +918,6 @@ RouteAction getRouteAction() {
       return routeAction;
     }
 
-    // TODO(chengyuanzhang): delete and do not use after routing feature is always ON.
```

Comment:
This might be irrelevant change...  This PR cleans up some `PathMatcher`/`RouteMatch` APIs, which are used here. So I am just deleting this unused code.